src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -338,32 +338,82 @@ internal static string ToDebugString(this BindingRestrictions restrictions)
 
     internal static class DynamicMetaObjectBinderExtensions
     {
-        internal static DynamicMetaObject DeferForPSObject(this DynamicMetaObjectBinder binder,
-                                                           params DynamicMetaObject[] args)
+        internal static DynamicMetaObject DeferForPSObject(this DynamicMetaObjectBinder binder, DynamicMetaObject arg0, bool moreRestrictionForArg0 = false)
```

Comment:
For consistency, `arg0` should be renamed `target` because we only care about the target.

I also think restrictions are specific to COM so the optional parameter could be named `targetIsComObject`.